### DISCUSSION TITLE
getUrl() on media collection not working

### DISCUSSION THREAD

**Discussion by Ikechukwu-Ezugworie** [spatie/laravel-medialibrary#2573] [Q&A]
I have a block of code that attaches the media to all model in a collection,  
but the ``` getUrl() ``` method,  throws a ```Call to a member function getUrl() on string ```

this is the code:
```php
$managementPage = ManagementPage::get();
        foreach ($managementPage as $key => $value) {
            $managementPage[$key]->getFirstMediaUrl()->getUrl();
        }
        return response()->json([
            'data' => $managementPage
        ]);
```
here's the error:
```php
 "message": "Call to a member function getUrl() on string",
    "exception": "Error",
    "file": "/Users/ik/Documents/Dev/City Gate Estate Web/admin-backend/app/Http/Controllers/ManagementPageController.php",
    "line": 20,
```

any help guys 🙏  on how to retrieve the media url on a collection

**Comment by miclf** [comment]
`->getFirstMediaUrl()` already returns a URL, as a string.

As a result, no method can be called after it, like you do in your example with `->getFirstMediaUrl()->getUrl()`.
